feat(sql): add refresh script for t_bibliographic_metadata generation

camillernd opened the pull request:
## 🤔 What

* Added a **SQL script** to **fill and refresh** `t_bibliographic_metadata`.
* The script truncates the table and **recomputes all bibliographic metadata** from existing data in `t_document` and related tables.
* The script is named:

  ```
  sql/4_refresh_bibliographic_metadata.sql
  ```
---

## 🤷‍♂️ Why

* This table is essential for OAI-PMH and Z39.50 services as it stores precomputed metadata for harvesting.

---

## 🔍 How

* The script first **truncates** `t_bibliographic_metadata` to remove stale records.
* It runs a single `INSERT INTO ... SELECT` statement aggregating:

  * Titles and descriptions from `t_description`
  * Authors and contributors from `t_caver`
  * Publisher names from `t_name` (via `t_grotto`)
  * Languages, coverages, subjects, and file paths
  * Identifiers with prefixes (url:, doi:, etc)
  * Derived types (image, text, etc.)
---
## 🧪 Testing

* The script has been tested successfully in PgAdmin with a dump containing **226,944** documents.

* Execution time observed: 10–15 seconds.

camillernd committed:
feat(sql): add refresh script for t_bibliographic_metadata generation

bsoufflet commented on sql/4_refresh_bibliographic_metadata.sql:
Peux tu me donner les arguments pour ne pas avoir choisi la solution de la vue materialisée ?

Si cela te semble possible de passer en mode materialized view alors il faudrait : 
- modifier le fichier 4_materialized_views.sql en ajoutant la vue à la fin avec `CREATE MATERIALIZED VIEW v_bibliographic_metadata AS` et `WITH NO DATA;` à la fin
(il va falloir changer le nom de la table mais c'est juste pour respecter nos conventions)
- modifier les endroits où on appelle t_bibliographic_metadata au lieu de v_bibliographic_metadata maintenant
- modifier 4_refresh_views.sql pour y ajouter le `REFRESH MATERIALIZED VIEW v_bibliographic_metadata;`
- supprimer le fichier sql/4_refresh_bibliographic_metadata.sql

camillernd replied:
Je n’avais pas choisi la vue matérialisée au départ principalement pour des raisons de simplicité pendant le développement : Yohann avait déjà préparé le modèle Sails sur la table t_bibliographic_metadata, ce qui permettait de travailler plus vite avec le mock et de tester les endpoints.

Je pense qu’une vue matérialisée sera effectivement préférable à terme.

Quelle solution vous semble la plus pertinente ?

1) Passer tout de suite à la vue matérialisée, modifier ce qu’a fait Yohann, puis une fois que c’est validé, je commence l’implémentation des endpoints du serveur OAI directement sur la vue.

2) Finir d’abord l’implémentation des endpoints en testant avec le mock de t_bibliographic_metadata, mettre en place le module correspondant sur le serveur OAI et m’assurer qu'il communique bien avec l’API, et migrer ensuite vers la vue matérialisée une fois que tout est fonctionnel.

bsoufflet replied:
Autant passer directement à la vue materialisée dans ta PR. le changement sur le code de Yohann est vraiment minime (juste un changement de nom).
Derrière tout fonctionnera comme avant et on pourra facilement faire un refresh de la table et vérifier le bon fonctionnement de ta query d'update :)

camillernd committed:
refactor(sql): migrate bibliographic metadata to materialized view

- Replace t_bibliographic_metadata table with v_bibliographic_metadata materialized view
- Update 4_materialized_views.sql to create the view
- Update 4_refresh_views.sql to refresh the view
- Remove obsolete 4_refresh_bibliographic_metadata.sql script
- Update Sails model TBibliographicMetadata to point to the view

camillernd commented:
**Changes:**

* Added `CREATE MATERIALIZED VIEW` (`WITH NO DATA` ) in `4_materialized_views.sql`
* Added `REFRESH MATERIALIZED VIEW` in `4_refresh_views.sql`
* Removed the obsolete `4_refresh_bibliographic_metadata.sql` script
* Updated the Sails model `TBibliographicMetadata` to point to the view

**Tested:**

* View successfully created and populated with **mock data**
* API endpoints tested locally with valid IDs from the mock dataset
* Tested the following endpoint:

  ```
  GET /api/v1/bibliographic-metadata/:id/format/:format
  ```

  Example request:

  ```
  GET /api/v1/bibliographic-metadata/5555/format/marc21-fr
  ```
* Records returned correctly in MARC format

bsoufflet commented on api/models/TBibliographicMetadata.js:
Il ne serait pas préférable de renommer aussi le fichier en VBibliographicMetadata.js ?
